Client/Frontend/Home/ReaderPanel.swift

```diff
@@ -250,87 +253,124 @@ class ReadingListPanel: UITableViewController, HomePanel, SWTableViewCellDelegat
-                    make.left.equalTo(welcomeLabel.snp_left)
-                    make.width.equalTo(ReadingListPanelUX.WelcomeScreenItemWidth)
-                })
-
-                let readerModeImageView = UIImageView(image: UIImage(named: "ReaderModeCircle"))
-                containerView.addSubview(readerModeImageView)
-                readerModeImageView.snp_makeConstraints({ (make) -> Void in
-                    make.centerY.equalTo(readerModeLabel)
-                    make.right.equalTo(welcomeLabel.snp_right)
-                })
-
-                let readingListLabel = UILabel()
-                containerView.addSubview(readingListLabel)
-                readingListLabel.text = NSLocalizedString("Save pages to your Reading List by tapping the book plus icon in the Reader View controls.", comment: "See http://mzl.la/1LXbDOL")
-                readingListLabel.font = ReadingListPanelUX.WelcomeScreenItemFont
-                readingListLabel.textColor = ReadingListPanelUX.WelcomeScreenItemTextColor
-                readingListLabel.numberOfLines = 0
-                readingListLabel.snp_makeConstraints({ (make) -> Void in
-                    make.top.equalTo(readerModeLabel.snp_bottom).offset(ReadingListPanelUX.WelcomeScreenPadding)
-                    make.left.equalTo(welcomeLabel.snp_left)
-                    make.width.equalTo(ReadingListPanelUX.WelcomeScreenItemWidth)
-                })
-
-                let readingListImageView = UIImageView(image: UIImage(named: "AddToReadingListCircle"))
-                containerView.addSubview(readingListImageView)
-                readingListImageView.snp_makeConstraints({ (make) -> Void in
-                    make.centerY.equalTo(readingListLabel)
-                    make.right.equalTo(welcomeLabel.snp_right)
-                })
-
-                containerView.snp_makeConstraints({ (make) -> Void in
-                    // Let the container wrap around the content
-                    make.top.equalTo(logoImageView.snp_top)
-                    make.bottom.equalTo(readingListLabel.snp_bottom)
-                    make.left.equalTo(welcomeLabel).offset(ReadingListPanelUX.WelcomeScreenItemOffset)
-                    make.right.equalTo(welcomeLabel).offset(ReadingListPanelUX.WelcomeScreenCircleOffset)
-                    // And then center it in the overlay view that sits on top of the UITableView
-                    make.center.equalTo(overlayView)
-                })
+    func firefoxAccountChanged(notification: NSNotification) {
+        if notification.name == NotificationFirefoxAccountChanged {
+            if let result = profile.readingList?.getAvailableRecords() where result.isSuccess {
+                records = result.successValue
+
+                if records?.count == 0 {
+                    tableView.scrollEnabled = false
+                    if let overlayView = emptyStateOverlayView {
```

Comment:
Nit: Instead of having this conditional logic to check if it's added and create it not created, you can make emptyStateOverlayView a lazy variable and add the view setup code inline like so:

```swift
private lazy var emptyStateOverlay: UIView = {
    // the codez
}()
```